src/metar/metar_csv2ioda.py

```diff
@@ -0,0 +1,322 @@
+
+    def __init__(self, filename, date):
+
+        self.filename = filename
+        self.date = date
+        self.meteo_utils = meteo_utils.meteo_utils()
+
+        # Read in CSV-formatted file of METAR data
+        self._rd_metars()
+
+        return
+
+    def _rd_metars(self):
+
+        # data is the dictionary of incoming observation (METAR) data
+        data = {}
+
+        data['ob_icao'] = []
+        data['ob_lat'] = []
+        data['ob_lon'] = []
+        data['ob_time'] = []
+        data['ob_datetime'] = []
+        data['ob_elev'] = []
+        data['ob_psfc'] = []
+        data['ob_temp'] = []
+        data['ob_spfh'] = []
+        data['ob_uwnd'] = []
+        data['ob_vwnd'] = []
+
+        '''
+        Read in the METARs data
+        Header contains: ICAO, Unix_time, Latitude, Longitude, Elev, MSLP, Altimeter, Temp, Dewp, Wdir, Wspd
+        '''
+
+        # open file in read mode
+        with open(self.filename, 'r') as fh:
+            # pass the file object to reader() to get the reader object
+            csv_dict_reader = DictReader(fh)
+            column_names = csv_dict_reader.fieldnames
+            print(column_names)
+            # Iterate over each row in the csv using reader object
+            for row in csv_dict_reader:
+                # row variable is a list that represents a row in csv
+
+                if row['ICAO'] is '':
+                    continue
+                else:
+                    icao = str(row['ICAO'])
+                try:
+                    utime = int(row['Unix_time'])
+                except:
+                    continue
+                try:
+                    lat = float(row['Latitude'])
+                except:
+                    continue
+                try:
+                    lon = float(row['Longitude'])
+                except:
+                    continue
+                try:
+                    elev = float(row['Elev'])
+                except:
+                    continue
+                try:
+                    temp = float(row['Temp']) + self.meteo_utils.C_2_K
+                except:
+                    temp = netCDF4.default_fillvals['f4']
+                try:
+                    dewp = float(row['Dewp']) + self.meteo_utils.C_2_K
+                except:
+                    dewp = netCDF4.default_fillvals['f4']
+                try:
+                    wdir = float(row['Wdir'])
+                except:
+                    wdir = netCDF4.default_fillvals['f4']
+                try:
+                    wspd = float(row['Wspd']) * self.meteo_utils.KTS_2_MS
+                except:
+                    wspd = netCDF4.default_fillvals['f4']
+
+                if ( (wdir is not netCDF4.default_fillvals['f4']) and (wspd is not netCDF4.default_fillvals['f4']) ):
+                    if (wdir==0 and wspd==0):
+                        uwnd = 0.0
+                        vwnd = 0.0
+                    elif (wdir>0 and wdir<=360 and wspd>0):
```

Comment:
wdir will never be exactly zero but it will be 360? Just double checking.

Reply:
This one is actually tricky and extremely particular to my own METAR processing for how I stored the data prior to reformatting to CSV file(s).  At NCAR-RAL, I created a MySQL db of METARs.  I made absolutely sure that a true north wind is stored as 360 degrees.  That was done quite purposefully to permit the wind direction to be zero degrees in the event the raw METAR reported a string like "VRB003" meaning variable wind direction at 3 knots.  This type of observation isn't DA-useful, because we cannot create unambiguous  east/northward_wind variables.  However, my reason for permitting it in MySQL and the post-created CSV file(s) is to consider a day in the distant future where a "VRB" wind direction could potentially be useful to DA.  So this is a bit of edge case just due to how I stored the data and created my own CSV files.

Reply:
Ok just checking. That is tricky.